From: Ben Hutchings <ben@decadent.org.uk>
Date: Wed, 4 Oct 2017 02:46:53 +0100
Subject: i40e: Build for 64-bit targets only

i40e now uses cmpxchg64(), which only works on a 64-bit (or non-SMP)
system.  It seems reasonable to expect that 40G hardware is not used
on 32-bit systems.

Fixes: 841c950d67c6 ("i40e/i40evf: use cmpxchg64 when updating private ...")
Signed-off-by: Ben Hutchings <ben@decadent.org.uk>
---
 drivers/net/ethernet/intel/Kconfig | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/net/ethernet/intel/Kconfig b/drivers/net/ethernet/intel/Kconfig
index 1feb54b6d92e..dd17bdf5561e 100644
--- a/drivers/net/ethernet/intel/Kconfig
+++ b/drivers/net/ethernet/intel/Kconfig
@@ -215,6 +215,7 @@ config I40E
 	tristate "Intel(R) Ethernet Controller XL710 Family support"
 	imply PTP_1588_CLOCK
 	depends on PCI
+	depends on 64BIT
 	---help---
 	  This driver supports Intel(R) Ethernet Controller XL710 Family of
 	  devices.  For more information on how to identify your adapter, go
